phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java

```diff
@@ -255,6 +257,7 @@ public static void setEndTaskStatus(PhoenixConnection connForTask, Task.TaskReco
             Task.addTask(connForTask, taskRecord.getTaskType(), taskRecord.getTenantId(), taskRecord.getSchemaName(),
                     taskRecord.getTableName(), taskStatus, data, taskRecord.getPriority(),
                     taskRecord.getTimeStamp(), endTs, true);
+            connForTask.commit();
```

Comment:
nit: null check for `connForTask`

Reply:
Also, this and the previous change in this file seem outside the scope of this JIRA which is just to fix the flaky test.